lib/core/test/unit/Cardano/Wallet/Primitive/AddressDiscovery/SequentialSpec.hs

```diff
@@ -596,7 +597,7 @@ deriving instance Arbitrary a => Arbitrary (ShowFmt a)
 
 instance Arbitrary AddressPoolGap where
     shrink _ = []
-    arbitrary = arbitraryBoundedEnum
+    arbitrary = mkUnboundedAddressPoolGap <$> choose (10, 20)
```

Comment:
Hi @KtorZ 

Just out of curiosity, what's the reason for the upper limit of `20` here?

Is it to keep the running time of tests within a reasonable limit?

Reply:
> Is it to keep the running time of tests within a reasonable limit?

Precisely. Remember that the gap serves as an upper bound for how many addresses we need to generate. Generating 100k addresses take roughly 30s on my machine. For a property test that runs thousands of scenarios,  that is not acceptable. 

Plus, having such large gap is not recommended (I wanted to actually add a disclaimer in the swagger doc which I'll do right away) so we can't really (nor are willing to) provide any strong guarantees about performances under such conditions. "Use at your own risks".